calicoctl/calico_ctl/container.py

```diff
@@ -506,7 +506,8 @@ def get_ip_and_pool(ip):
             ip = result[1][0]
         pool = get_pool_or_exit(ip)
     elif ip is not None and '/' in ip:
-        pool = IPPool(ip)
+        cidr = IPNetwork(ip)
+        pool = client.get_pool(IPAddress(cidr.first))
```

Comment:
If you specify a CIDR that does not exist, the pool returned will be None.  I think we actually want to just exit here if the pool does not exit.

I would replace client.get_pool() with the get_pool_or_exit() which will exit with an error if the pool does not exist.